framework/src/modules/chain/logic/transaction_pool.js

```diff
@@ -42,1058 +63,220 @@ const __private = {};
- * @param {Transactions} transactions - Transactions module instance
- * @param {Loader} loader - Loader module instance
- */
-TransactionPool.prototype.bind = function(accounts, transactions, loader) {
-	modules = {
-		accounts,
-		transactions,
-		loader,
-	};
-
-	jobsQueue.register(
-		'transactionPoolNextBundle',
-		nextBundle,
-		self.bundledInterval
-	);
-
-	jobsQueue.register(
-		'transactionPoolNextExpiry',
-		nextExpiry,
-		self.expiryInterval
-	);
-};
 
-/**
- * Returns true if index exists in at least one lists of indexes.
- * Lists: unconfirmed, bundled, queued, multisignature.
- *
- * @param {string} id - Transaction id
- * @returns {boolean}
- */
-TransactionPool.prototype.transactionInPool = function(id) {
-	return [
-		self.unconfirmed.index[id],
-		self.bundled.index[id],
-		self.queued.index[id],
-		self.multisignature.index[id],
-	].some(index => typeof index === 'number');
-};
-
-/**
- * Gets an unconfirmed transaction based on transaction id.
- *
- * @param {string} id - Transaction id
- * @returns {Object|undefined} Transaction or undefined
- */
-TransactionPool.prototype.getUnconfirmedTransaction = function(id) {
-	const index = self.unconfirmed.index[id];
-	return self.unconfirmed.transactions[index];
-};
-
-/**
- * Gets a bundled transaction based on transaction id.
- *
- * @param {string} id - Transaction id
- * @returns {Object|undefined} Transaction or undefined
- * @todo This function is never called
- */
-TransactionPool.prototype.getBundledTransaction = function(id) {
-	const index = self.bundled.index[id];
-	return self.bundled.transactions[index];
-};
-
-/**
- * Gets a queued transaction based on transaction id.
- *
- * @param {string} id - Transaction id
- * @returns {Object|undefined} Transaction or undefined
- */
-TransactionPool.prototype.getQueuedTransaction = function(id) {
-	const index = self.queued.index[id];
-	return self.queued.transactions[index];
-};
-
-/**
- * Gets multisignature transactions based on transaction id.
- *
- * @param {string} id - Transaction id
- * @returns {Object[]} Array of multisignature transactions
- */
-TransactionPool.prototype.getMultisignatureTransaction = function(id) {
-	const index = self.multisignature.index[id];
-	return self.multisignature.transactions[index];
-};
-
-/**
- * Gets unconfirmed transactions based on limit and reverse option.
- *
- * @param {boolean} reverse - Reverse order of results
- * @param {number} limit - Limit applied to results
- * @returns {Object[]} Of bundled transactions
- */
-TransactionPool.prototype.getUnconfirmedTransactionList = function(
-	reverse,
-	limit
-) {
-	return __private.getTransactionList(
-		self.unconfirmed.transactions,
-		reverse,
-		limit
-	);
-};
+	/**
+	 * Bounds input parameters to private and sets up the liskhq-transaction-pool module.
+	 *
+	 * @param {Transactions} processTransactions - Transactions processing module instance
+	 */
+	bind(processTransactions) {
+		modules = {
```

Comment:
Wouldn't it be better to avoid using a module variable and use a class property instead? That way it's simple to test/mock

Reply:
Sure! I was refactoring code and kept modules variable as defined in the old implementation. I will change it.